### ISSUE TITLE
Support passing regexes to `find()` queries

### ISSUE THREAD

**Issue by muety** [go-kivik/kivik#537]
It would be cool to pass a `regexp.Regexp` in `find()` calls, i.e. something like this:

```go
rows, err := s.db.Find(context.TODO(), map[string]interface{}{
    "selector": map[string]interface{}{
        "title": map[string]interface{}{
	     "$regex": regexp.MustCompile(`foo`),
        },
    },
})
```

Right now, the regex selector seems to be ignored and all docs are returned instead.

**Comment by flimzy** [comment]
That's an interesting proposal. It got me looking at the `regexp.Regexp` type--it satisfies the `fmt.Stringer` interface, but not the `json.Marshaler` interface, which would be needed to support this out of the box.  It might be nice if this support were added, but it might be considered a breaking change.

I filed an issue with the Go project. Let's see what they say.  Although I doubt they'll add this functionality soon, any discussion can be informative nonetheless.

https://github.com/golang/go/issues/46159

**Comment by flimzy** [comment]
As the upstream proposal has now been accepted, I think it's safe to say it would be inappropriate to add (duplicate) support to kivik, so I'll be closing this issue.  Of course I don't know when that proposal will actually be implemented, but I suggest following that issue for updates, rather than this one.

**Comment by flimzy** [comment]
It looks like this change should finally be included in Go 1.21.
